wurst/systems/Hints.wurst

```diff
@@ -0,0 +1,109 @@
+package Hints
+import ClosureTimers
+import LinkedList
+import ChatCommands
+import Game
+
+let TIME_BETWEEN_HINTS = 60.0
+let HINTS_OFF_MESSAGE = GENERAL_COLOR + "Your hints are now " + GOLD_COLOR + "off" + GENERAL_COLOR + "."
+let HINTS_ON_MESSAGE = GENERAL_COLOR + "Your hints are now " + GOLD_COLOR + "on" + GENERAL_COLOR + "."
+
+class Hint
+    
+    static force players = CreateForce()
+
+    private string message
+    private trigger action = null
+
+    construct(string hint)
+        this.message = hint
+
+    construct(string hint, code what)
+        this.message = hint
+        this.action = CreateTrigger()
+        TriggerAddAction(this.action, what)
+
+    static function pingUnit(unit which)
+        pingLoc(GetUnitLoc(which))
+
+    static function pingLoc(location loc)
+        PingMinimapLocForForceEx(players, loc, 5, bj_MINIMAPPINGSTYLE_FLASHY, 100, 100, 100)
+
+    static function addPLayer(player p)
+        ForceAddPlayer(players, p)
+        printTimedToPlayer(HINTS_ON_MESSAGE, 5, p)
+
+    static function removePlayer(player p)
+        ForceRemovePlayer(players, p)
+        printTimedToPlayer(HINTS_OFF_MESSAGE, 5, p)
+
+    static function togglePlayer(player p)
+        if (IsPlayerInForce(p, players))
+            Hint.removePlayer(p)
+        else    
+            Hint.addPLayer(p)
+
+    function display()
+        if IsPlayerInForce(GetLocalPlayer(), players)
+            printTimedToPlayer(GOLD_COLOR + "Hint: |r" + GENERAL_COLOR + message + "|r", 10, GetLocalPlayer())
+            if action != null
+                TriggerExecute(action)
+
+    static function initialize()
+        for i = 0 to 11
+            ForceAddPlayer(players, Player(i))
+               
+        registerCommandAll("h") (triggerPlayer, args) -> 
+            Hint.togglePlayer(triggerPlayer)
+        
+public function displayNextHint(LLIterator<Hint> iterator)
+    let hint = iterator.next()
+    hint.display()
+    if iterator.hasNext() == false
+        iterator.reset()
+        
+public function doPeriodicDisplay(LinkedList<Hint> hintList)
+        let iterator = hintList.iterator()
+        doPeriodically(TIME_BETWEEN_HINTS, (cb) -> displayNextHint(iterator))
+
+public function checkStat(unit troll)
```

Comment:
I don't think this belongs in the package for hints. Put it in the stat decay or its own package.

Reply:
I am not really ok with this suggestion, I put it in this package because I thought it would be convenient to use the hint instance to check the player in force in case they disabled the hints.

Reply:
@Jaccouille I think its fine for now, we can always move it later. Perhaps you can periodically check every `gameConfig.getStatLossInterval()` and set the message duration to the same so that you'll only have one such message on your screen at a time (excluding the fade time). Might be a bit annoying but we can tweak it later.